### PR TITLE
Add `tribal_land_frac`

### PR DESCRIPTION
**Pull request by bendnorman**
This PR adds a `tribal_land_percent` column to the county_fips and county_wide and county_long tables which is the percent of the county within tribal land.

@TrentonBush I'm a little confused about adding the column to the data mart tables. Seems like there is a lot of columns adding and dropping happening.

### CODE REVIEW

**Review by TrentonBush — CHANGES_REQUESTED** [review]

File: src/dbcp/etl.py [hunk 1]
```diff
@@ -108,8 +108,7 @@ def etl_ncsl_state_permitting() -> Dict[str, pd.DataFrame]:
 
 def etl_fips_tables() -> Dict[str, pd.DataFrame]:
     """Master state and county FIPS table ETL."""
-    source_path = Path("/app/data/data_cache/tl_2021_us_county.zip")
-    fips = dbcp.extract.fips_tables.extract(census_path=source_path)
+    fips = dbcp.extract.fips_tables.extract()
```
[the remark this answers is not in the document]
Comment (bendnorman) [reply src/dbcp/etl.py:111]: Ah good point. I forgot that we'll probably need to update this data!

File: src/dbcp/metadata/data_mart.py [hunk 2]
```diff
@@ -155,6 +156,12 @@
     Column("unprotected_land_area_km2", Float),
     Column("federal_fraction_unprotected_land", Float),
     Column("county_land_area_km2", Float),
+    Column(
+        "tribal_land_percent",
+        Float,
+        CheckConstraint("tribal_land_percent >= 0.0 AND tribal_land_percent <= 1.0"),
```
[the remark this answers is not in the document]
Comment (bendnorman) [reply src/dbcp/metadata/data_mart.py:162]: Groovy!

File: src/dbcp/transform/fips_tables.py [hunk 3]
```diff
@@ -2,12 +2,59 @@
...
+    )
+    counties["tribal_land_percent"] = (
+        counties["tribal_land_intersection"] / counties["land_area_km2"]
+    )
+
+    tribal_land_larger_than_county = counties.tribal_land_percent > 1
+    logger.info(
+        f"Number of counties with where tribal land is great 100% of land area:\n{tribal_land_larger_than_county.value_counts()}"
+    )
+    counties["tribal_land_percent"] = counties.tribal_land_percent.mask(
```
[the remark this answers is not in the document]
Comment (bendnorman) [reply src/dbcp/transform/fips_tables.py:44]: Good call. Added `raw_tribal_land_frac`.

File: src/dbcp/transform/fips_tables.py [hunk 4]
```diff
@@ -2,12 +2,59 @@
...
+    tribal_land = tribal_land.to_crs("ESRI:102008")
+
+    dissolved_tribal = tribal_land.dissolve()
+    dissolved_tribal_geometry = dissolved_tribal.geometry.iloc[0]
+
+    # Calculate intersection, convert to km
+    counties["tribal_land_intersection"] = (
+        counties.intersection(dissolved_tribal_geometry).area / 1e6
+    )
+    counties["tribal_land_percent"] = (
```
[the remark this answers is not in the document]
Comment (bendnorman) [reply src/dbcp/transform/fips_tables.py:36]: updated 👍

**Pull request merged**